### PR TITLE
added pattern test cases for F, G and H alphabet detectors

### PR DESCRIPTION
**Pull request by kashifkhan0771**
### Description:
This PR add the pattern test cases for detectors starting with alphabet `f` to `h`. It also separate the integration test cases for those detectors.

### Test Output:
pattern-f-h-tests-01
pattern-f-h-tests-02
pattern-f-h-tests-03
pattern-f-h-tests-04

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

### CODE REVIEW

File: pkg/detectors/github_oauth2/github_oauth2.go [hunk 1]
```diff
@@ -20,6 +19,9 @@ type Scanner struct {
 
 // Ensure the Scanner satisfies the interfaces at compile time.
 var _ detectors.Detector = (*Scanner)(nil)
+var _ detectors.EndpointCustomizer = (*Scanner)(nil)
```
Comment (kashifkhan0771) [inline pkg/detectors/github_oauth2/github_oauth2.go:22]: Added this because a engine test case was failing due to missing method in this detector.

Comment (ahrav) [reply pkg/detectors/github_oauth2/github_oauth2.go:22]: hey @kashifkhan0771 would you mind double checking some comments [here](https://github.com/trufflesecurity/trufflehog/pull/3587#issuecomment-2471533293) around this. I might've misunderstood Miccah's comment and tried to fix this issue with: https://github.com/trufflesecurity/trufflehog/pull/3592

Not sure if I made it worse 😓

Comment (ahrav) [reply pkg/detectors/github_oauth2/github_oauth2.go:22]: Thank you for double checking. ❤️

**Review by ahrav — APPROVED** [review]

File: pkg/detectors/github_oauth2/github_oauth2.go (hunk 1 above)
Comment (kashifkhan0771) [reply pkg/detectors/github_oauth2/github_oauth2.go]: Done 🚀 Updated with your changes from main.

**Pull request merged**